### ISSUE TITLE
Add api to list smart modules

### ISSUE THREAD

**Issue by morenol** [infinyon/fluvio-client-node#259]

**Comment by EstebanBorai** [comment]
Hi @morenol!

I'm checking on this issue and as I understand we want to achieve the same result as from running `fluvio smart-module list` via CLI but through the NodeJS client am I right?

**Comment by EstebanBorai** [comment]
I'm working on this here: https://github.com/infinyon/fluvio-client-node/pull/262, still a WIP while I get the `SmartModuleSpec` struct into JavaScript to be able to expose it in the client side.

**Comment by morenol** [comment]
Yes, you can check the list_smartmodules function in the wasm client. In the admin.rs. Probably a create_smartmodule and delete_smartmodule functions would be great for e2e testing

**Comment by EstebanBorai** [comment]
> Yes, you can check the list_smartmodules function in the wasm client. In the admin.rs. Probably a create_smartmodule and delete_smartmodule functions would be great for e2e testing

Totally agree! I'm checking on SmartModule creation as well! Perhaps we should land three features. So it would be more like SmartModule CRUD support.
